crates/language/src/buffer.rs

```diff
@@ -209,11 +209,17 @@ pub async fn prepare_completion_documentation(
     documentation: &lsp::Documentation,
     language_registry: &Arc<LanguageRegistry>,
     language: Option<Arc<Language>>,
+    maximum_documentation_length: usize,
 ) -> Documentation {
     match documentation {
         lsp::Documentation::String(text) => {
             if text.lines().count() <= 1 {
-                Documentation::SingleLine(text.clone())
+                let mut text = text.clone();
+                if text.len() > maximum_documentation_length {
+                    text.truncate(maximum_documentation_length - 3);
```

Comment:
I believe this will panic if the byte offset `maximum_documentation_length` is not at a valid UTF8 character boundary. There's a function in the codebase called `util::truncate_and_trailoff` that handles this properly.